src/database/sqlx-mysql/src/mysql_connection_pool.rs

```diff
@@ -0,0 +1,39 @@
+// Copyright Kamu Data, Inc. and contributors. All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+use database_common::DatabaseError;
+use dill::{component, scope, Singleton};
+use sqlx::{MySql, MySqlPool, Transaction};
+
+/////////////////////////////////////////////////////////////////////////////////////////
+
+pub struct MySqlConnectionPool {
+    mysql_pool: MySqlPool,
+}
+
+#[component(pub)]
+#[scope(Singleton)]
+impl MySqlConnectionPool {
```

Comment:
Have you considered not wrapping `sqlx` types? The likelihood is that we will either never change the underlying library, or changing to other lib will require significant modifications to all wrappers ... so we might as well use `sqlx` types directly.

Reply:
This is still unclear. Mainly, we have to work on error handling strategy a bit more to answer this.

Reply:
Okay, managed to inline. Initially this had more responsibilities, but not anymore.